mos-platform/common/c/math.cc

```diff
@@ -0,0 +1,22 @@
+#include <math.h>
+
+template <typename T> static inline bool _is_nan(const T x) { return x != x; }
```

Comment:
May as well actually implement the macro in `<math.h>`:

`#define isnan(arg) __builtin_isfpclass(arg, __FPCLASS_SNAN | __FPCLASS_QNAN)`

Reply:
Or use the existing `__builtin_isnan`?

Reply:
Also, I now notice that `__builtin_fmin` and `__builtin_fmax` along with several other math functions are already defined and seem to be correct. Perhaps better to use these? Are these covered by existing tests in the separate test suite?

Reply:
Yeah, `__builtin_isnan` lowers to the same thing and is clearer.

`__builtin_fmin` doesn't help us; it lowers to a compiler intrinsic that most naturally lowers to a call to `fmin`. By comparison, `__builtin_isnan` lowers to a call to `__unordsf2` in `compiler_rt`, plus a bit of additional logic.